ui/shared/Revision.jsx

```diff
@@ -47,9 +47,40 @@ export class Revision extends React.PureComponent {
 
     this.state = {
       clipboardVisible: false,
+      ready: false,
+      tooltipOpen: false,
     };
+    this.spanRef = React.createRef();
   }
 
+  componentDidMount() {
+    this.setReady();
+  }
+
+  componentDidUpdate() {
+    this.setReady();
+  }
+
+  setReady() {
+    if (this.spanRef.current) {
+      this.setState({
+        ready: true,
+      });
+    }
+  }
+
+  handleTooltip = async () => {
+    this.handleTooltipOpen();
+  };
+
+  handleTooltipOpen = () => {
```

Comment:
You might consider using an `UncontrolledToolTip` instead.  That way you don't have to handle the opening and closing of it.  I think this might help with the clipboard icon being left behind sometimes as well.